File: platform-kotlin-mobile/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmobile/resource/KotlinAndroidResources.java
package ru.pocketbyte.locolaser.platform.kotlinmobile.resource;

import ru.pocketbyte.locolaser.platform.kotlinmobile.resource.file.KotlinAndroidResourceFile;
import ru.pocketbyte.locolaser.resource.file.ResourceFile;

import java.io.File;
import java.util.Set;

public class KotlinAndroidResources extends AbsKotlinImplementationPlatformResources {

    private String mAppPackage;

    public KotlinAndroidResources(File dir, String name, String interfaceName, String appPackage) {
        super(dir, name, interfaceName);
        mAppPackage = appPackage;
    }

    @Override
    protected ResourceFile[] getResourceFiles(Set<String> locales) {
        return new ResourceFile[] {
                new KotlinAndroidResourceFile(getFile(),
                        this.className, this.classPackage,
                        this.interfaceName, this.interfacePackage, mAppPackage)
        };
    }
}

File: platform-kotlin-mobile/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmobile/resource/KotlinAndroidResources.kt
package ru.pocketbyte.locolaser.platform.kotlinmobile.resource

import ru.pocketbyte.locolaser.platform.kotlinmobile.resource.file.KotlinAndroidResourceFile
import ru.pocketbyte.locolaser.resource.file.ResourceFile

import java.io.File

class KotlinAndroidResources(
        dir: File,
        name: String,
        interfaceName: String,
        private val mAppPackage: String
) : AbsKotlinImplementationPlatformResources(dir, name, interfaceName) {

    override fun getResourceFiles(locales: Set<String>): Array<ResourceFile> {
        return arrayOf(KotlinAndroidResourceFile(file,
                this.className, this.classPackage,
                this.interfaceName, this.interfacePackage, mAppPackage))
    }
}
